src/server/server.js

```diff
@@ -74,29 +55,33 @@ app.use((_, res, next) => {
 app.get('/isAlive', (req, res) => res.send('alive'))
 app.get('/isReady', (req, res) => res.send('ready'))
 
-app.get('/login',
-  passport.authenticate('azuread-openidconnect', { failureRedirect: '/error', 'session': false })
-)
-
-app.post('/callback',
-  passport.authenticate('azuread-openidconnect', { "session": false, failWithError: true }),
-  (req, res) => {
-    console.log('callback')
-    console.log(req.body)
-    res.cookie('speil', `${req.authInfo.id_token}`, { secure: true })
-    res.cookie('spade', `${req.authInfo.access_token}`, { httpOnly: true, secure: true })
-    res.redirect('/')
-  }, (err, req, res, next) => {
-        console.log("i dont know what happened")
-        console.log(err)
-        return res.json(err);
+app.get('/login', (req, res) => {
+   req.session.nonce = generators.nonce()
+   const url = azureClient.authorizationUrl({
+      scope: config.oidc.scope,
+      redirect_uri: config.oidc.redirectUrl,
+      response_type: config.oidc.responseType,
+      response_mode: 'form_post',
+      nonce: req.session.nonce
     })
+    res.redirect(url)
+})
 
-app.get('/logout', (req, res) => {
-   req.session.destroy(_ => {
-      req.logOut()
-      res.redirect(config.destroySessionUrl)
-   })
+app.post('/callback', (req, res) => {
+   const params = azureClient.callbackParams(req)
+   const nonce = req.session.nonce
+   azureClient.callback(config.oidc.redirectUrl, params, { nonce })
+      .then((tokenSet) => {
+         res.cookie('speil', `${tokenSet['id_token']}`, { httpOnly: true })
+         res.cookie('spade', `${tokenSet['access_token']}`, { httpOnly: true })
```

Comment:
for øyeblikket så funker det nok å sette access tokenet i cookie, men på sikt tror jeg en mer farbar vei er å sette denne i session, også kjøre alle kall til spade via speil sin backend. Dvs. at alle kall til `speil.nais.adeo.no/api/*` proxes til `http://spade/*`, men med `Authorization: Bearer <token>` satt i requesten som proxes videre. Da "slipper" vi å støtte CORS i spade, og spade slipper også å eksponere en ingress.

Reply:
Det er planen ja, måtte bare gjenopprette status quo med nytt bibliotek først.